salt/postgres/files/postgres-backup-to-s3.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+#
+# Backup PostgreSQL databases and upload to AWS S3
+
+set -euo pipefail
+
+PSQL_ADMIN_USER="postgres"
+
+# shellcheck disable=SC1091
+. /home/sysadmin-tools/aws-settings.local
+
+export AWS_ACCESS_KEY_ID
+export AWS_SECRET_ACCESS_KEY
+export AWS_DEFAULT_REGION
+
+if [ "$LOGNAME" != "postgres" ]; then
+    echo "ERROR: Execution of $0 stopped as not run by user postgres!"
+    exit 2
+fi
+
+if [ ! -x "$AWS_CLI" ]; then
+    echo "Error: The aws executable is not installed"
+    exit 3
+fi
+
+# Create work directory resolving permissions alerts when running psql from root.
+WORKDIR=$(mktemp -d)
+chown $PSQL_ADMIN_USER:$PSQL_ADMIN_USER "$WORKDIR"
+
+mapfile -t DATABASES < <(cd "$WORKDIR" && sudo -u $PSQL_ADMIN_USER /usr/bin/psql -t --csv -c 'select datname from pg_database')
```

Comment:
Instead of the `WORKDIR` workaround, why not `su - postgres -c "command"` ?

Reply:
Fixed in new commit.